concepts/admission-webhook-cert-management/README.md

```diff
@@ -0,0 +1,145 @@
+  ```yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: serving-cert
+  namespace: kyma-system
+spec:
+  # $(SERVICE_NAME) and $(SERVICE_NAMESPACE) must be substituted by the webhook service/namespace
+  dnsNames:
+  - $(SERVICE_NAME).$(SERVICE_NAMESPACE).svc
+  - $(SERVICE_NAME).$(SERVICE_NAMESPACE).svc.cluster.local
+  issuerRef:
+    kind: Issuer
+    name: selfsigned-issuer
+  secretName: webhook-server-cert
+```
+
+In addition to that, it would mark it's webhook configuration with a special anntotation to let the `ca injector` automatically populate the `caBundle`. See [more](https://cert-manager.io/docs/concepts/ca-injector/).
+
+  ```yaml
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: validation.webhook.telemetry.kyma-project.io
+  annotations:
+    cert-manager.io/inject-ca-from: kyma-system/serving-cert
+...  
+```
+
+This approach has a few downsides:
+* `cert-manager` is deployed using Helm, so the chart has to be maintained (upgrades, security patches, etc.)
+* deploying the `cert-manager` as a standalone Kyma component means opening it's API to Kyma customers. It also means that we should build expertise within Kyma, some team should take ownership of this nedw component, etc.
+* we only neeed support of self-signed certificates and certificate rotation, which is a very small subset of `cert-manager`'s features
+
+### <a name="init-container"></a>Init container
+
+https://www.velotio.com/engineering-blog/managing-tls-certificate-for-kubernetes-admission-webhook
+
+This is slightly modified version of [one of the currently used approaches](#a-nameserver-codeagenerate-the-ca-cert-and-the-server-cert-and-update-the-webhook-configuration-in-the-webhook-server-code-itself). Instead of implementing the logic in the webhook server code, it's packaged as a Docker image and run as an init container.
+
+### <a name="component-operator"></a>Component operator
+
+With [the new modularization strategy](https://github.com/kyma-project/community/tree/main/concepts/modularization) we could implement certificate management as a part of component operators. The component operators can also watch and rotate certificates.
+The certificate management functionality can be also packaged as a Go module and shared between the operators.
+
+## Conclusion
+
+After some discussions we came to the conclsuion that even though `cert-manager` is a recommended solution, it would be an overkill for our use case. Self-signed certificate generation and rotation is a tiny subset of all `cert-manager` capabilities and can be implemented in future component operators as a part of the modularization startegy.
```

Comment:
```suggestion
After some discussions, we came to the conclusion that even though `cert-manager` is a recommended solution, it would be an overkill for our use case. Self-signed certificate generation and rotation is a tiny subset of all `cert-manager` capabilities and can be implemented in future component operators as a part of the modularization strategy.
```